features/element-hiding/main.js

```diff
@@ -0,0 +1,128 @@
+    { id: 'closest-empty-single-nested-div', description: 'closest-empty rule: parent of target element is hidden', hidden: true },
+    { id: 'closest-empty-siblings-nested-div', description: 'closest-empty rule: parent of target element with empty siblings is hidden', hidden: true },
+    { id: 'closest-empty-siblings-content-nested-div', description: 'closest-empty rule: parent of target element with non-empty siblings is not hidden', hidden: false },
+    { id: 'closest-empty-frame-content', description: 'closest-empty rule: parent of element containing iframe is not hidden', hidden: false },
+    { id: 'closest-empty-delayed-frame', description: 'closest-empty rule: parent of element that loads 3p frame after delay is not hidden', hidden: false },
+    { id: 'closest-empty-intermediate-element', description: 'closest-empty rule: only outermost element should be hidden when target is nested', hidden: false },
+    { id: 'closest-empty-delayed-parent', description: 'closest-empty rule: parent of empty element added to page after delay is hidden', hidden: true },
+    { id: 'override-basic', description: 'override rule: element is not hidden when override rule present', hidden: false }
+];
+
+const results = {
+    page: 'element-hiding',
+    date: (new Date()).toUTCString(),
+    results: []
+};
+
+const delayedSelectors = [
+    { id: 'hide-delayed', class: 'hide-test', parent: 'hide' },
+    { id: 'hide-empty-delayed', class: 'hide-empty-test', parent: 'hide-empty' },
+    { id: 'closest-empty-delayed', class: 'closest-empty-test', parent: 'closest-empty-delayed-parent' }
+];
+
+function init () {
+    // inject iframes
+    loadFrames();
+
+    // inject frames and other content after delay
+    injectDelayedContent();
+
+    // wait 2s then check results
+    setTimeout(checkResults, 2000);
+}
+
+function checkResults () {
+    for (const testCase of expectedResults) {
+        const element = document.getElementById(testCase.id);
+        const isHidden = element.style.getPropertyValue('display') === 'none';
+        const result = testCase.hidden === isHidden;
+        updateTable(testCase.id, testCase.description, result);
+    }
+}
+
+function updateTable (name, description, testPassed) {
+    const table = document.getElementById('results-table');
+    const row = table.insertRow(-1);
+
+    const descriptionCell = row.insertCell(0);
+    const testCell = row.insertCell(1);
+
+    // set default values and colors
+    descriptionCell.innerText = description;
+
+    const result = {
+        id: name,
+        hiddenProperly: testPassed
+    };
+
+    if (testPassed) {
+        testCell.innerText = 'Success';
+        testCell.style.backgroundColor = '#71bf69';
+    } else {
+        testCell.innerText = 'Failed';
+        testCell.style.backgroundColor = '#f97268';
+    }
+
+    results.results.push(result);
+}
+
+function injectDelayedContent () {
+    setTimeout(function () {
```

Comment:
```suggestion
    setTimeout(() => {
```

(we use arrow function mostly everywhere now)

Reply:
Old habits die hard 😢